### ISSUE TITLE
Default resource scanners ignore resourcePackages when resourceClasses is populated

### ISSUE THREAD

**Issue by hectorlf** [swagger-api/swagger-core#3504]
E.g. https://github.com/swagger-api/swagger-core/blob/master/modules/swagger-jaxrs2/src/main/java/io/swagger/v3/jaxrs2/integration/JaxrsAnnotationScanner.java

Line 65:
```
if (openApiConfiguration.getResourceClasses() != null && !openApiConfiguration.getResourceClasses().isEmpty()) {
   ... some classpath scanning ...
   return output;
}
```

What is the purpose of this behavior? Shouldn't I be able to specify a bunch of packages and **also** a set of classes that I don't want to load by package, and get them all merged together?

**Comment by frantuma** [comment]
Currently packages and classes are indeed mutually exclusive, it probably would make sense to support also the behaviour you mention; this can be achieved by providing your own [custom scanner](https://github.com/swagger-api/swagger-core/wiki/Swagger-2.X---Integration-and-configuration#resource-scanning), this could be also added to the swagger-core provided scanners in case you feel like opening a PR.

**Comment by hectorlf** [comment]
I've created a pull request, but while working on it I've realized this has a chance to break back-compatibility.

@frantuma, the custom scanner worked fine for my use case, so thanks for that.

**Comment by frantuma** [comment]
Thanks for your effort, and yes the idea was to have a new scanner class (hierarchy) with the desired behaviour to add to the already existing (and used by default) ones, maybe adapting base `GenericOpenApiScanner` with a flag. In case it would be great if you could adapt the PR

**Comment by hectorlf** [comment]
Sorry, I'm afraid I fail to come up with meaningful class names that aren't confusing or simply garbage. All suggestions welcome!

Also, a flag in GenericOpenApiScanner or any of the other scanners won't work either, because not all usages can pass in flags (e.g. my original issue was related to the Gradle plugin).

Maybe the way forward is to leave this change for the next major version of swagger-core? I'll risk repeating that this PR brings Swagger closer to how Jersey handles package scanning.

**Comment by frantuma** [comment]
@hectorlf yes, as behaviour can be tailored with custom scanners, this is not terribly urgent, thanks a lot for  your effort.

For future reference / contributors, the idea of the flag is to adjust behaviour in `GenericOpenApiScanner` and `JaxrsAnnotationScanner` depending on  a newly added configuration (and gradle/maven plugin paramerter) boolean property like `scanBothConfigClassesAndPagackes`  (or hopefully a better name..) defaulting to current behaviour `false`; if set to `true` both classes and packages would be scanned; in next major version default can be set to `true`, as it makes more sense and is more similar to Jersey behaviour as you mentioned.

**Comment by MichakrawSB** [comment]
Closing this issue as part of our https://github.com/swagger-api/swagger-core/issues/5074.
This issue is outdated and/or inactive for a long time, and we’re not able to revisit it meaningfully.

If this problem is still relevant, please open a new issue and reference this one for context.
Thank you for your understanding!
